### PR TITLE
feat: Table support cssVar

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Table support cssVar |
| 🇨🇳 Chinese | feat: Table 组件支持 cssVar |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at ceb83ed</samp>

The pull request improves the table component styles by using new utility functions and design tokens for unit conversion and calculation. It also refactors some CSS properties and files to enhance readability and maintainability. It updates the `scripts/check-cssinjs.tsx` file to fix some minor issues. The affected files are `components/table/style/*.ts` and `scripts/check-cssinjs.tsx`.

### <samp>🤖 Generated by Copilot at ceb83ed</samp>

*  Use `unit` and `calc` functions from `@ant-design/cssinjs` to handle unit conversion and calculation in table style files ([link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-af9a50095f1ce23e8cabb4de4623f4b2e62fd9942a8b26e3c0084df3baef1fedL1-R17),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-af9a50095f1ce23e8cabb4de4623f4b2e62fd9942a8b26e3c0084df3baef1fedL22-R33),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-af9a50095f1ce23e8cabb4de4623f4b2e62fd9942a8b26e3c0084df3baef1fedL91-R108),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-af9a50095f1ce23e8cabb4de4623f4b2e62fd9942a8b26e3c0084df3baef1fedL152-R162),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-67a1e8dee753b6957c495e69a9f52ff716bc8217d0d810e812fafc472006c422L1-R5),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-67a1e8dee753b6957c495e69a9f52ff716bc8217d0d810e812fafc472006c422L26-R35),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-67a1e8dee753b6957c495e69a9f52ff716bc8217d0d810e812fafc472006c422L63-R71),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-67a1e8dee753b6957c495e69a9f52ff716bc8217d0d810e812fafc472006c422L118-R126),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-67a1e8dee753b6957c495e69a9f52ff716bc8217d0d810e812fafc472006c422L145-R154),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e26ffdd87a4e231c5eb74ff7bbc7f4b540c50e210e9427c9a4ac4f4bd76170d8R1-R2),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e26ffdd87a4e231c5eb74ff7bbc7f4b540c50e210e9427c9a4ac4f4bd76170d8L33-R40),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e26ffdd87a4e231c5eb74ff7bbc7f4b540c50e210e9427c9a4ac4f4bd76170d8L52-R58),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e26ffdd87a4e231c5eb74ff7bbc7f4b540c50e210e9427c9a4ac4f4bd76170d8L96-R101),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e26ffdd87a4e231c5eb74ff7bbc7f4b540c50e210e9427c9a4ac4f4bd76170d8L105-R110),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e26ffdd87a4e231c5eb74ff7bbc7f4b540c50e210e9427c9a4ac4f4bd76170d8L147-R152),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L1-R2),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288R15),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L40-R42),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L58-R60),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L75-R77),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L96-R100),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L104-R110),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L117-R125),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-8a902239be08dcc1c4526d6cf665a5f97bf9fab9877016c14d5a17a513c0d288L125-R135),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL1-R5),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL214-R214),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL221-R221),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cR236),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL248-R251),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL260-R262),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL266-R268),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL281-R283),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL287-R289),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL340-R345),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL368-R370),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL445-R516),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cL459-R531),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-eac67a8e6855fba3c427add14286587a235c994bef6e7c4cc0a79bb71da691cdL1-R12),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b12ced1b5a8e3d92028e94a37af829fa284362fad0900262a089dd77ec7bb47cL1-R2),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b12ced1b5a8e3d92028e94a37af829fa284362fad0900262a089dd77ec7bb47cL12-R13),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b12ced1b5a8e3d92028e94a37af829fa284362fad0900262a089dd77ec7bb47cL47-R48),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b96dfeee25155ede756ec06496950ba9bd9e106f79d05547329cb758b2a9bb4cL1-R1),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b96dfeee25155ede756ec06496950ba9bd9e106f79d05547329cb758b2a9bb4cR20-R21),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b96dfeee25155ede756ec06496950ba9bd9e106f79d05547329cb758b2a9bb4cL28-R44),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-b96dfeee25155ede756ec06496950ba9bd9e106f79d05547329cb758b2a9bb4cL74-R83),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e079a40882207691b3e8205df7d39d92e7730ca90d195b3a5668b5bb0ff4901eL1-R7),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e079a40882207691b3e8205df7d39d92e7730ca90d195b3a5668b5bb0ff4901eL25-R36),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e079a40882207691b3e8205df7d39d92e7730ca90d195b3a5668b5bb0ff4901eL39-R45),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-e079a40882207691b3e8205df7d39d92e7730ca90d195b3a5668b5bb0ff4901eL48-R51),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-688d7b5bf4ed6f2776e4c06ce857493222dd4e438e3ae9c98e7a5d28d8cf0127L1-R2),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-688d7b5bf4ed6f2776e4c06ce857493222dd4e438e3ae9c98e7a5d28d8cf0127L15-R20),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-688d7b5bf4ed6f2776e4c06ce857493222dd4e438e3ae9c98e7a5d28d8cf0127L29-R33),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-c6838065243b7ac4061f054c164666558ae753e8686b72d93c1d86fa20fda524L1-R8),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-c6838065243b7ac4061f054c164666558ae753e8686b72d93c1d86fa20fda524L14),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-c6838065243b7ac4061f054c164666558ae753e8686b72d93c1d86fa20fda524L23-R23),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-7c7346bcc5b9d2329966a268db0784b15a13add22938c2fe06c9efead085b3fdL1-R1),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-7c7346bcc5b9d2329966a268db0784b15a13add22938c2fe06c9efead085b3fdL7-R9),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-7c7346bcc5b9d2329966a268db0784b15a13add22938c2fe06c9efead085b3fdL33-R33),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-7c7346bcc5b9d2329966a268db0784b15a13add22938c2fe06c9efead085b3fdL57-R57))
* Add `prepareComponentToken` function to `components/table/style/index.ts` to extract common logic of generating default token values ([link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cR378-R441),[link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-a9612177ff7fc2e50de3a66777bcee3c89711d02a709c61366efbc3b838a8e6cR483))
* Remove `table` component from `checkCSSVar` function in `scripts/check-cssinjs.tsx` as it is no longer needed ([link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-5b054b9f47063e9b12b347b6e9bd669ad645226477f1d56fbdbc20afa54e109eL68))
* Add `React` import statement to `scripts/check-cssinjs.tsx` to allow rendering components with `ConfigProvider` ([link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-5b054b9f47063e9b12b347b6e9bd669ad645226477f1d56fbdbc20afa54e109eR2))
* Move `React` import statement to the top of `scripts/check-cssinjs.tsx` and `generateCssinjs` import statement to the bottom, to follow convention ([link](https://github.com/ant-design/ant-design/pull/45856/files?diff=unified&w=0#diff-5b054b9f47063e9b12b347b6e9bd669ad645226477f1d56fbdbc20afa54e109eL10-R14))

### CODE REVIEW

File: components/table/style/expand.ts [hunk 1]
```diff
@@ -115,9 +119,11 @@ const genExpandStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
       },
 
       [`${componentCls}-row-indent + ${componentCls}-row-expand-icon`]: {
-        marginTop:
-          (fontSize * lineHeight - lineWidth * 3) / 2 -
-          Math.ceil((fontSizeSM * 1.4 - lineWidth * 3) / 2),
```
Comment (li-jia-nan) [inline components/table/style/expand.ts:120]: 这里的 `Math.ceil` 不知道怎么写

File: components/table/style/virtual.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ const genVirtualStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
             position: 'sticky',
             insetInlineStart: 0,
             overflow: 'hidden',
-            width: `calc(var(--virtual-width) - ${token.lineWidth}px)`,
+            width: `calc(var(--virtual-width) - ${unit(lineWidth)})`,
```
Comment (li-jia-nan) [inline components/table/style/virtual.ts:33]: 这里不知道会不会有问题

File: components/table/style/fixed.ts [hunk 3]
```diff
@@ -72,7 +74,7 @@ const genFixedStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
           position: 'absolute',
           top: 0,
           bottom: 0,
-          zIndex: zIndexTableSticky + 1,
+          zIndex: calc(zIndexTableSticky).add(1).equal(),
```
Comment (MadCcc) [inline components/table/style/fixed.ts:77]:
```suggestion
          zIndex: calc(zIndexTableSticky).add(1).equal({unit: false}),
```

File: components/table/style/fixed.ts [hunk 4]
```diff
@@ -93,15 +95,19 @@ const genFixedStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
           position: 'relative',
 
           '&::before': {
-            boxShadow: `inset 10px 0 8px -8px ${shadowColor}`,
+            boxShadow: `inset ${unit(10)} 0 ${unit(8)} ${unit(
```
Comment (MadCcc) [inline components/table/style/fixed.ts:98]: 变量才需要用 unit

Comment (li-jia-nan) [reply components/table/style/fixed.ts]: 改回去了

**Comment by li-jia-nan** [comment]
样式没生效，argos 居然没检查出来，奇怪

**Comment by li-jia-nan** [comment]
argos diff 有问题是因为 demo 里面用到了 Badge，等 Badge 迁移完毕之后就好了

File: components/table/style/expand.ts [hunk 5]
```diff
@@ -60,11 +62,11 @@ const genExpandStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
         height: expandIconSize,
         padding: 0,
         color: 'inherit',
-        lineHeight: `${expandIconSize}px`,
+        lineHeight: unit(expandIconSize),
         background: tableExpandIconBg,
         border: tableBorder,
         borderRadius,
-        transform: `scale(${checkboxSize / expandIconSize})`,
+        transform: `scale(${calc(checkboxSize).div(expandIconSize).equal()})`,
```
Comment (MadCcc) [inline components/table/style/expand.ts:69]: 这个应该和 radio 一个问题，带 px 计算了

File: components/table/style/fixed.ts [hunk 6]
```diff
@@ -114,15 +116,19 @@ const genFixedStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
           position: 'relative',
 
           '&::after': {
-            boxShadow: `inset -10px 0 8px -8px ${shadowColor}`,
+            boxShadow: `inset ${unit(calc(10).mul(-1).equal())} 0 ${unit(8)} ${unit(
```
Comment (MadCcc) [inline components/table/style/fixed.ts:119]: 纯数字就不要用 unit 和 calc 了，没意义

File: components/table/style/index.ts [hunk 7]
```diff
@@ -442,11 +518,11 @@ export default genComponentStyleHook(
       tableHeaderSortHoverBg: headerSortHoverBg,
       tableHeaderIconColor: baseColorAction
         .clone()
-        .setAlpha(baseColorAction.getAlpha() * opacityLoading)
+        .setAlpha(calc(baseColorAction.getAlpha()).mul(opacityLoading).equal())
         .toRgbString(),
       tableHeaderIconColorHover: baseColorActionHover
```
Comment (MadCcc) [inline components/table/style/index.ts:523]: 颜色计算也是无效的，需要放到 component token 里去

File: components/table/style/index.ts [hunk 8]
```diff
@@ -456,12 +532,12 @@ export default genComponentStyleHook(
       tableSelectedRowBg: rowSelectedBg,
       tableSelectedRowHoverBg: rowSelectedHoverBg,
       zIndexTableFixed,
-      zIndexTableSticky: zIndexTableFixed + 1,
+      zIndexTableSticky: calc(zIndexTableFixed).add(1).equal(),
```
Comment (MadCcc) [inline components/table/style/index.ts:535]:
```suggestion
      zIndexTableSticky: calc(zIndexTableFixed).add(1).equal({unit: false}),
```
记得去单位

Comment (MadCcc) [reply components/table/style/index.ts:535]: 话说这个就没必要用 calc，没有 token 参与运算

**Review by MadCcc — APPROVED** [review]

**Pull request merged**